### ISSUE TITLE
Searching for children of already expanded nodes breaks expansion after removing search term

### ISSUE THREAD

**Issue by paxuclus** [neos/neos-ui#2583]
### Description

When searching for a node that is on the third level inside of the node tree, upon removing the search term, the parent node on the first level cannot be expanded again.

### Steps to Reproduce

1. Configure nodeTree loadingDepth to be 1
```
Neos:
  Neos:
    userInterface:
      navigateComponent:
        nodeTree:
          loadingDepth: 1
```
2. Expand a node that has two levels of children
3. Open the search input
4. Search for a node that is two levels below the collapsed node
5. Remove search term
6. Expand the same node again

#### Expected behavior

The node should expand again, showing revealing all of its child nodes.

#### Actual behavior

Only the icon next to the node switches to the collapsed state, but no child nodes are visible.

See the following GIF for a demonstration using the demo:
![neos-pagetree-search](https://user-images.githubusercontent.com/15905038/69242207-a6d6f680-0ba0-11ea-9eab-acdeab33e596.gif)

### Affected Versions

Neos: 4.3.8 and 5.0.4

UI: 4.0.1 and 5.0.1

**Comment by dimaip** [comment]
Thanks for the detailed report! Looks buggy :)

**Comment by erkenes** [comment]
This bug is still existing in Neos 7.1 (UI version: v7.1.1)
